### ISSUE TITLE
Show parent Nav as active as well

### ISSUE THREAD

**Issue by coderchef26** [mui/toolpad#4538]
### Discussed in https://github.com/mui/toolpad/discussions/4534

<div type='discussions-op-text'>

<sup>Originally posted by **gazi786** December 12, 2024</sup>
If a dependent route is selected how can I make the parent nav active as well. That is have the Mui-selected class attached to the parent node as well.</div>

Also goes in a parent link has a grandchild all three should be shown active.

![image](https://github.com/user-attachments/assets/8ca38e93-ec62-4c43-89f0-8c37d7ba4bae)

This is more of a feature request rather then an issue.

**Search keywords**:

**Comment by Janpot** [comment]
This could probably benefit from a bit more discussion and benchmarking before we implement a solution? Personally I'm not convinced that making this configurable should be part of Toolpad  API. It feels like for this level of customizability, a slot for the whole navigation menu allows you to customize the whole thing as much as you want without burdening the library with extra complexity.

**Comment by gazi786** [comment]
Agreed... a slotProp to assign the Mui-selected class to the parents of the related selected child. Since the AppProvider's NavigationContext renders the navigation, I have tried to push a state through redux to the AppProvider but have been unsuccessful at the moment.

Still a learning curve for me.

The DashboardLayout component controls many actions for the dashboard. I'm trying to look into how the side nav gets rendered.

**Comment by gazi786** [comment]
Since the navigation is recursively rendered using the component:

    <DashboardSidebarSubNavigation
            subNavigation={navigation}
            onLinkClick={handleNavigationLinkClick}
            isMini={isMini}
            isFullyExpanded={isNavigationFullyExpanded}
            hasDrawerTransitions={hasDrawerTransitions}
            selectedItemId={selectedItemIdRef.current}
          />

in the DashboardLayout component...

So, would it be easy to add a prop, say, isParentActive, that adds the Mui-selected class to the Parent or even use the selectedItemId reference?

**Comment by prakhargupta1** [comment]
I checked a few tools and docs that have nested navigation and found that most of them don't show the parent as selected. Two of those who do this:

1. https://e2b.dev/docs
2. Google Analytics (but Gmail doesn't :p)
<img width="227" alt="Screenshot 2024-12-17 at 1 06 13 PM" src="https://github.com/user-attachments/assets/0b348c9c-97da-4b3e-a578-4ba4ccb1f4b2" />

As per the linked discussion, @gazi786 managed to resolve this for their use case. I have added 'waiting for upvotes' label to see if more users want this.

**Comment by gazi786** [comment]
@prakhargupta1 thats what I have checked.. Alot of the apps don't show the parent Nav active. It usually suits when we working with top navigation. 

If we get a good number of feedback perhaps we can include it in a later release. 

Thanks for the awesome feedback and support.

**Comment by prakhargupta1** [comment]
> Thanks for the awesome feedback and support.

@gazi786 Thanks for explaining the issue well. From the screenshots you shared, I can see that the dashboard is coming along well. In case you need any help or discuss a problem with Toolpad Core, feel free to [schedule a call](https://calendly.com/prakhar-mui/30mins).

**Comment by gazi786** [comment]
Thanks mate will do...
